provider/openstack/upgrades.go

```diff
@@ -72,9 +72,19 @@ type tagExistingSecurityGroupsStep struct {
 	env *Environ
 }
 
+var tagSecGroupRetryStrategy = retry.CallArgs{
+	Clock:       clock.WallClock,
+	MaxDuration: 30 * time.Second,
+	Delay:       time.Second,
+	BackoffFunc: retry.ExpBackoff(time.Second, 10*time.Second, 1.5, true),
+}
+
 func (t tagExistingSecurityGroupsStep) buildReplaceTagsWithRetry(ctx context.ProviderCallContext, neutronClient NetworkingNeutron, groupId string, tags []string) retry.CallArgs {
-	retryStrategy := shortRetryStrategy
+	retryStrategy := tagSecGroupRetryStrategy
 	retryStrategy.IsFatalError = func(err error) bool {
+		if strings.Contains(err.Error(), "TLS handshake timeout") {
```

Comment:
I think it's ok to retry on any error as we can't be sure what is transient or not (or can we?). We will backoff and eventually hit the max duration and then will error out anyway.